### PR TITLE
8285306: Fix typos in java.desktop

### PR DESCRIPTION
**Pull request by magicus**
I ran `codespell` on the `src/java.desktop` directory, and accepted those changes where it indeed discovered real typos.

I ignored typos in public methods and variables. Maybe they can be fixed later on without much fanfare, if they are in internal classes. Typos in exposed APIs are likely here to stay.

I will update copyright years using a script before pushing (otherwise like every second change would be a copyright update, making reviewing much harder).

The long term goal here is to make tooling support for running `codespell`. The trouble with automating this is of course all false positives. But before even trying to solve that issue, all true positives must be fixed. Hence this PR.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8285306](https://bugs.openjdk.org/browse/JDK-8285306): Fix typos in java.desktop

### Reviewers
 * [Andrey Turbanov](https://openjdk.org/census#aturbanov) (@turbanoff - Committer) ⚠️ Review applies to [98e635a5](https://git.openjdk.org/jdk/pull/8328/files/98e635a544ecab4600719e3096e078624fdfe4e0)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * @SWinxy (no known github.com user name / role) ⚠️ Review applies to [2850610d](https://git.openjdk.org/jdk/pull/8328/files/2850610dc70ec9db8f1d23c72cf5e6889f823fa3)

### Contributors
 * Andrey Turbanov `<aturbanov@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/8328/head:pull/8328` \
`$ git checkout pull/8328`

Update a local copy of the PR: \
`$ git checkout pull/8328` \
`$ git pull https://git.openjdk.org/jdk pull/8328/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8328`

View PR using the GUI difftool: \
`$ git pr show -t 8328`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/8328.diff">https://git.openjdk.org/jdk/pull/8328.diff</a>

</details>

### CODE REVIEW

**Comment by prrace** [comment]
static class AquaHierarchyButtonListener implements HierarchyListener {
-        // Everytime a hierarchy is change we need to check if the button if moved on or from
+        // Every time a hierarchy is change we need to check if the button if moved on or from

change -> changed
if moved -> is moved

src/java.desktop/macosx/classes/sun/lwawt/macosx/CEmbeddedFrame.java


-            // it won't be invoced if focuse is moved to a html element
+            // it won't be invoced if focus is moved to a html element

invoced -> invoked
"a html" -> "an html"

src/java.desktop/macosx/native/libjsound/PLATFORM_API_MacOSX_MidiOut.c
+    // $$fb 2002-04-04: It is responsibility of the application developer to

"is the"

src/java.desktop/windows/native/libjsound/PLATFORM_API_WinOS_MidiOut.c

+    // $$fb 2002-04-04: It is responsibility of the application developer to

same as above

Regarding changes in gif + freetype 
diff --git a/src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c b/src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c

diff --git a/src/java.desktop/share/native/libfontmanager/freetypeScaler.c b/src/java.desktop/share/native/libfontmanager/freetypeScaler.c

Please exclude ALL 3rd party libraries from this PR.

File: src/java.desktop/windows/classes/sun/awt/windows/WPrinterJob.java [hunk 1]
```diff
@@ -386,7 +386,7 @@ public WPrinterJob()
     }
 
     /* Implement DisposerTarget. Weak references to an Object can delay
-     * its storage reclaimation marginally.
+     * its storage reclamation marginally.
      * It won't make the native resources be release any more quickly, but
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/classes/sun/awt/windows/WPrinterJob.java:390]:
```suggestion
     * It won't make the native resources be released any more quickly, but
```

File: src/java.desktop/windows/native/libawt/windows/awt_Label.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ class AwtLabel : public AwtComponent {
     virtual MsgRouting WmEraseBkgnd(HDC hDC, BOOL& didErase);
 
     /*
-     * if WM_PAINT was recieving when we can not paint
+     * if WM_PAINT was receiving when we can not paint
      * then setup m_needPaint end when can call this function
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awt_Label.h:62]:
```suggestion
     * if WM_PAINT was received when we cannot paint
     * then setup m_needPaint and when can paint call this function
```

File: src/java.desktop/windows/native/libawt/windows/awt_PrintDialog.cpp [hunk 3]
```diff
@@ -54,7 +54,7 @@ LRESULT CALLBACK PrintDialogWndProc(HWND hWnd, UINT message,
             if ((LOWORD(wParam) == IDOK) ||
                 (LOWORD(wParam) == IDCANCEL))
             {
-                // If we recieve on of these two notifications, the dialog
+                // If we receive on of these two notifications, the dialog
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awt_PrintDialog.cpp:57]:
```suggestion
                // If we receive one of these two notifications, the dialog
```

File: src/java.desktop/windows/native/libawt/windows/awt_PrintJob.cpp [hunk 4]
```diff
@@ -3170,7 +3170,7 @@ LRESULT CALLBACK PageDialogWndProc(HWND hWnd, UINT message,
             if ((LOWORD(wParam) == IDOK) ||
                 (LOWORD(wParam) == IDCANCEL))
             {
-                // If we recieve on of these two notifications, the dialog
+                // If we receive on of these two notifications, the dialog
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awt_PrintJob.cpp:3173]:
```suggestion
                // If we receive one of these two notifications, the dialog
```

File: src/java.desktop/windows/native/libawt/windows/awt_Toolkit.cpp [hunk 5]
```diff
@@ -1083,7 +1083,7 @@ LRESULT CALLBACK AwtToolkit::WndProc(HWND hWnd, UINT message,
       // Special awt message to call Imm APIs.
       // ImmXXXX() API must be used in the main thread.
       // In other thread these APIs does not work correctly even if
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awt_Toolkit.cpp:1085]:
```suggestion
      // In other threads these APIs do not work correctly even if
```

File: src/java.desktop/windows/native/libawt/windows/awt_Toolkit.cpp [hunk 6]
```diff
@@ -1083,7 +1083,7 @@ LRESULT CALLBACK AwtToolkit::WndProc(HWND hWnd, UINT message,
...
-      // it returs with no error. (This restriction is not documented)
+      // it returns with no error. (This restriction is not documented)
       // So we must use thse messages to call these APIs in main thread.
```
Comment (aivanov-jdk) [inline src/java.desktop/windows/native/libawt/windows/awt_Toolkit.cpp:1087]:
```suggestion
      // So we must use these messages to call these APIs in main thread.
```

File: src/java.desktop/share/classes/javax/swing/GroupLayout.java [hunk 7]
```diff
@@ -1651,7 +1651,7 @@ int getSpringSize(Spring spring, int axis, int type) {
 
         /**
          * Used to compute how the two values representing two springs
-         * will be combined.  For example, a group that layed things out
+         * will be combined.  For example, a group that laid things out
```
[the remark this answers is not in the document]
Comment (magicus) [reply src/java.desktop/share/classes/javax/swing/GroupLayout.java:1654]: If you're not sure, I prefer to leave it as it is.

File: src/java.desktop/share/classes/sun/swing/text/TextComponentPrintable.java [hunk 8]
```diff
@@ -769,7 +769,7 @@ private void calculateRowsMetrics() {
                             && (y != previousY || height != previousHeight)) {
                         /*
                          * we do not store the same value as previous. in our
-                         * documents it is often for consequent positons to have
+                         * documents it is often for consequent positions to have
```
[the remark this answers is not in the document]
Comment (magicus) [reply src/java.desktop/share/classes/sun/swing/text/TextComponentPrintable.java:772]: Possibly. I suggest you file a follow up PR to address additional documentation errors.

**Comment by magicus** [comment]
@aivanov-jdk While I approve of finding and fixing spelling mistakes, grammatical or semantic errors, the mistakes you posted suggestions for was not at all associated with the mechanical changes made by codespell that this PR included, and I assume you noticed them just because you read the text around the typo my PR fixed. 

In hindsight, it would have been better if you've made these changes yourself, in a separate PR. 

But since you had spent the effort of looking for these problems, and suggesting fixes, I have now accepted all of them. Due to how the Github UI works, this was quite tedious, so please refrain from submitting more than a handful suggestions next time you review a PR.

**Comment by aivanov-jdk** [comment]
Thank you, @magicus,  I should've noted the changes and created a branch on top yours.

That's right. Even looking through all the changes in this PR was tedious, more than hundred files doesn't work well.

File: src/java.desktop/share/classes/sun/swing/text/TextComponentPrintable.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/sun/swing/text/TextComponentPrintable.java:772]: I've submitted [JDK-8286759](https://bugs.openjdk.java.net/browse/JDK-8286759): TextComponentPrintable: consequent -> consecutive positions.

**Comment by aivanov-jdk** [comment]
> Please exclude ALL 3rd party libraries from this PR.

Are these the only files to be excluded?

```
src/java.desktop/share/native/libfreetype/src/autofit/afcjk.c
src/java.desktop/share/native/libfreetype/src/autofit/aflatin.c
src/java.desktop/share/native/libfreetype/src/base/ftsynth.c
src/java.desktop/share/native/libfreetype/src/raster/ftraster.c
```

**Comment by magicus** [comment]
Thank you bot. I'm on it!

File: src/java.desktop/share/classes/javax/swing/JTree.java [hunk 9]
```diff
@@ -3310,45 +3310,45 @@ private int[] getModelIndexsForPath(TreePath path) {
...
+            int[]       indices = new int[count - 1];
             Object      parent = model.getRoot();
 
             for(int counter = 1; counter < count; counter++) {
-                indexs[counter - 1] = model.getIndexOfChild
+                indices[counter - 1] = model.getIndexOfChild
                                    (parent, path.getPathComponent(counter));
                 parent = path.getPathComponent(counter);
-                if(indexs[counter - 1] < 0)
+                if(indices[counter - 1] < 0)
```
Comment (turbanoff) [inline src/java.desktop/share/classes/javax/swing/JTree.java:3320]: let's add space after `if`

File: src/java.desktop/share/classes/javax/swing/JTree.java [hunk 10]
```diff
@@ -3310,45 +3310,45 @@ private int[] getModelIndexsForPath(TreePath path) {
...
      * Returns a <code>TreePath</code> created by obtaining the children
-     * for each of the indices in <code>indexs</code>. If <code>indexs</code>
+     * for each of the indices in <code>indices</code>. If <code>indices</code>
      * or the <code>TreeModel</code> is <code>null</code>, it will return
      * <code>null</code>.
      */
-    private TreePath getPathForIndexs(int[] indexs) {
-        if(indexs == null)
+    private TreePath getPathForIndexs(int[] indices) {
+        if(indices == null)
```
Comment (turbanoff) [inline src/java.desktop/share/classes/javax/swing/JTree.java:3335]: please add space after `if`

File: src/java.desktop/share/classes/javax/swing/JTree.java (hunk 9 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/JTree.java:3320]: Better not: if you look at the code around it, there's no space between a keyword and the opening parenthesis. So the code looks consistent. Changing this `if` will make the code inconsistent.

Comment (magicus) [reply src/java.desktop/share/classes/javax/swing/JTree.java:3320]: @turbanoff I'll go with @aivanov-jdk's suggestion here. Also, this is a PR created by an automation tool. I'd really prefer not to do any manual intervention, if possible, and at the very least not spelling related.

File: src/java.desktop/share/classes/javax/swing/JTree.java (hunk 10 above)
Comment (magicus) [reply src/java.desktop/share/classes/javax/swing/JTree.java:3335]: Same here. Space issues are not typos.

**Review by turbanoff — APPROVED** [review]

**Review by prrace — CHANGES_REQUESTED** [review]
We need to revert the native code changes to "libfreetype".
These are 3rd party imported files.
After that we should get this pushed and move on and I agree that no matter
the similarity this PR is TOO BIG. Which is I think mostly why it has taken this long and still not done.

**Comment by magicus** [comment]
I have now removed the changes to libfreetype, and also libjpeg. These were the only 3rd party source I could locate.

I apologize for this PR. It was too huge. If it is of any comfort, I've learnt my lesson and won't repeat it. And once it started getting pushed down by more prioritized work, I got  a bit scared about going back and look through all this for additional (unspecified) 3rd party sources, so it got left behind, hanging around.

@prrace I hope this is acceptable to merge now.

**Review by aivanov-jdk — COMMENTED** [review]
Is `src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c` a third-party code?

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m [hunk 11]
```diff
@@ -82,7 +82,9 @@ - (id) initWithJavaLayer:(jobject)layer
 
 - (void) blitTexture {
     if (self.ctx == NULL || self.javaLayer == NULL || self.buffer == nil || self.ctx.device == nil) {
-        J2dTraceLn4(J2D_TRACE_VERBOSE, "MTLLayer.blitTexture: uninitialized (mtlc=%p, javaLayer=%p, buffer=%p, device=%p)", self.ctx, self.javaLayer, self.buffer, ctx.device);
+        J2dTraceLn4(J2D_TRACE_VERBOSE,
+                    "MTLLayer.blitTexture: uninitialized (mtlc=%p, javaLayer=%p, buffer=%p, device=%p)", self.ctx,
+                    self.javaLayer, self.buffer, ctx.device);
```
[the remark this answers is not in the document]
Comment (magicus) [reply src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m:87]: This is from the merge. The only difference wrt to current code is "devide" => "device". I can change wrapping if you request it, but I'd prefer to only fix typos.

**Comment by magicus** [comment]
> Is `src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c` a third-party code?

I don't think so. I find no comments in the source code, or any kind of indication on this being so. If you say it is, I can revert the changes in that file.

**Comment by magicus** [comment]
I see now that Phil cryptically said:

> Regarding changes in gif + freetype
diff --git a/src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c b/src/java.desktop/share/native/libawt/awt/image/gif/gifdecoder.c \
\
diff --git a/src/java.desktop/share/native/libfontmanager/freetypeScaler.c b/src/java.desktop/share/native/libfontmanager/freetypeScaler.c \
\
Please exclude ALL 3rd party libraries from this PR.

That might be interpreted as stating that `gifdecoder.c` is 3rd party source code (although that was by no means clear to me the first time I read it). I'll revert the changes in that file.

File: src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m (hunk 11 above)
[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m:87]: > I'd prefer to only fix typos.

Let us keep only the typos.

**Comment by magicus** [comment]
@prrace I have now reverted the changes to glext.h, wsutils.h and multiVis.c. Is this finally ready for merging?

(Going forward, I think we absolutely need to have some way to document in the code tree that certain files is 3rd party, like the UPSTREAM notation I previously suggested, or some variant thereof. This "tribal knowledge" about what is 3rd party is not beneficial to anyone, and only wastes both your and my time...)

File: src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java [hunk 12]
```diff
@@ -65,7 +65,7 @@
  */
 public abstract class MetalTheme {
 
-    // Contants identifying the various Fonts that are Theme can support
+    // Constants identifying the various Fonts that a Theme can support
```
Comment (SWinxy) [inline src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java:68]: I think this is possibly meant to say "constants that *our* theme can support"?

File: src/java.desktop/share/classes/sun/awt/SunToolkit.java [hunk 13]
```diff
@@ -1431,7 +1431,7 @@ public IllegalThreadException() {
     private static final int MINIMAL_DELAY = 5;
 
     /**
-     * Parameterless version of realsync which uses default timout (see DEFAUL_WAIT_TIME).
+     * Parameterless version of realsync which uses default timeout (see DEFAULT_WAIT_TIME).
```
Comment (SWinxy) [inline src/java.desktop/share/classes/sun/awt/SunToolkit.java:1434]:
```suggestion
     * Parameterless version of realsync which uses the {@link #DEFAULT_WAIT_TIME default timeout}.
```

**Comment by SWinxy** [comment]
Everything looks fine, and you can freely ignore my suggestions. They are just thoughts. And yeah we should probably earmark third party files to avoid "tribal knowledge".

File: src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java (hunk 12 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java:68]: Please submit a new JBS issue for it if you can; if you can't, I can submit one on your behalf.

Possibly it meant to say “_the_ theme”.

File: src/java.desktop/share/classes/sun/awt/SunToolkit.java (hunk 13 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/sun/awt/SunToolkit.java:1434]: I'm for a clearer message:

Parameterless version of `{@code realsSync}` which uses _the_ default timeout of `{@link #DEFAULT_WAIT_TIME}`.

Let us address this in a separate issue.

File: src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java (hunk 12 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java:68]: I think the text is correct here: …that _**a** Theme_ can support

`MetalTheme` is a superclass for classes which implement Themes. Thus the constants define fonts which _a particular theme_ may use.

File: src/java.desktop/share/classes/sun/awt/SunToolkit.java (hunk 13 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/sun/awt/SunToolkit.java:1434]: [JDK-8294255](https://bugs.openjdk.org/browse/JDK-8294255): Add link to DEFAULT_WAIT_TIME in javadoc for SunToolKit.realsSync

Feel free to provide a patch.

File: src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java (hunk 12 above)
Comment (SWinxy) [reply src/java.desktop/share/classes/javax/swing/plaf/metal/MetalTheme.java:68]: Makes sense.

**Review by SWinxy — APPROVED** [review]

**Comment by magicus** [comment]
@prrace Do you think you can approve this now, so we can finally close it? (I promise I won't open huge PRs like this in the future; lesson well learnt.)

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**